apps/OpenSpace/ext/launcher/src/sgctedit/windowcontrol.cpp

```diff
@@ -0,0 +1,521 @@
+        {
+            _comboProjection->setCurrentIndex(ProjectionIndeces::Equirectangular);
+        }
+    }
+}
+
+template <typename T>
+void WindowControl::enableProjectionOption(T* comboModel, int selectionIndex, bool enable)
+{
+    auto* item = comboModel->item(selectionIndex);
+    if (item) {
+        item->setEnabled(enable);
+    }
+}
+
+void WindowControl::onMonitorChanged(int newSelection) {
+    _monIndex = newSelection;
+    if (_windowChangeCallback) {
+        _windowChangeCallback(_monIndex, _index, _windowDims);
+    }
+}
+
+void WindowControl::onProjectionChanged(int newSelection) {
+    _comboQuality->setVisible(newSelection != ProjectionIndeces::Planar);
+    _labelQuality->setVisible(newSelection != ProjectionIndeces::Planar);
+    _labelFovH->setVisible(newSelection == ProjectionIndeces::Planar);
+    _lineFovH->setVisible(newSelection == ProjectionIndeces::Planar);
+    _labelFovV->setVisible(newSelection == ProjectionIndeces::Planar);
+    _lineFovV->setVisible(newSelection == ProjectionIndeces::Planar);
+    _labelHeightOffset->setVisible(newSelection == ProjectionIndeces::Cylindrical);
+    _lineHeightOffset->setVisible(newSelection == ProjectionIndeces::Cylindrical);
+    _checkBoxSpoutOutput->setVisible(newSelection == ProjectionIndeces::Fisheye
+        || newSelection == ProjectionIndeces::Equirectangular);
+}
+
+void WindowControl::setDimensions(const QRectF& dimensions) {
+    _windowDims = dimensions;
+}
+
+void WindowControl::setWindowChangeCallback(
+                        std::function<void(int, int, const QRectF&)> cb)
+{
+    _windowChangeCallback = cb;
+}
+
+void WindowControl::setWebGuiChangeCallback(
+                        std::function<void(unsigned int)> cb)
+{
+    _windowGuiCheckCallback = cb;
+}
+
+void WindowControl::uncheckWebGuiOption() {
+    _checkBoxWebGui->setCheckState(Qt::Unchecked);
+}
+
+QRectF& WindowControl::dimensions() {
+    return _windowDims;
+}
+
+QLineEdit* WindowControl::lineEditSizeWidth() {
+    return _size_x;
+}
+
+QLineEdit* WindowControl::lineEditSizeHeight() {
+    return _size_y;
+}
+
+QLineEdit* WindowControl::lineEditSizeOffsetX() {
+    return _offset_x;
+}
+
+QLineEdit* WindowControl::lineEditSizeOffsetY() {
+    return _offset_y;
+}
+
+QCheckBox* WindowControl::checkBoxWindowDecor() {
+    return _checkBoxWindowDecor;
+}
+
+QCheckBox* WindowControl::checkBoxWebGui() {
+    return _checkBoxWebGui;
+}
+
+QCheckBox* WindowControl::checkBoxSpoutOutput() {
+    return _checkBoxSpoutOutput;
+}
+
+std::string WindowControl::windowName() {
+    return _windowName->text().toStdString();
+}
+
+sgct::ivec2 WindowControl::windowSize() {
+    return {
+        stoi(_size_x->text().toStdString()),
+        stoi(_size_y->text().toStdString())
+    };
+}
+
+sgct::ivec2 WindowControl::windowPos() {
+    return {
+        stoi(_offset_x->text().toStdString()),
+        stoi(_offset_y->text().toStdString())
+    };
+}
+
+bool WindowControl::isDecorated() {
```

Comment:
if all of these functions are still necessary, they can all be marked as `const`